### PR TITLE
DTrace build fix on Mac proposal.

### PR DESCRIPTION
**Pull request by devnexen**
The header generated comes with $ IDs thus breaking the build.
The probes are set with const address arguments already
which just add the qualifier again.

### CODE REVIEW

**Comment by dormando** [comment]
thanks! Fwiw I have no mac right now, so no way to validate this. Would appreciate anyone with a mac who could give it a quick validation/check.

**Comment by dormando** [comment]
Yeah, still need someone else with a Mac to come try this. Since you've been doing freebsd patches, I assume you've also tested on freebsd :)

**Comment by devnexen** [comment]
Thx for the reminder :-)

File: configure.ac [hunk 1]
```diff
@@ -177,6 +187,8 @@ if test "x$enable_dtrace" = "xyes"; then
     then
         dtrace_instrument_obj=yes
         rm conftest.h
+        # on Mac probe id are generated with $
+        CFLAGS="$CFLAGS -Wno-dollar-in-identifier-extension"
```
Comment (dormando) [inline configure.ac:191]: this is enabling for freebsd too?

**Comment by dormando** [comment]
I want to close this out but I don't have the resources to validate it myself; it looks like it's changing the build around a bit for darwin, which might overlap with other dtrace-enabled OS's.

Need someone else to +1 this. I can try again myself once I've gotten more of the new buildbots running as well, I guess. Either way I'd like to close this out in the next week as it's just hanging around.

**Comment by dormando** [comment]
I can see this build failure now. Will take a more careful look when I get a moment.

**Comment by dormando** [comment]
@devnexen rebase, address comment, and lets finish this? I have access to a recent mac so can validate. comment was about lack of DARWIN gating for the cflags change. maybe add some comments about why the other changes were done for darwin only?

**Comment by dormando** [comment]
Fails on my mac:
```
/usr/sbin/dtrace  -G -o memcached_dtrace.o -s ./memcached_dtrace.d memcached-memcached.o memcached-hash.o memcached-jenkins_hash.o memcached-murmur3_hash.o memcached-slabs.o memcached-items.o memcached-assoc.o memcached-thread.o memcached-daemon.o memcached-stats_prefix.o memcached-util.o memcached-cache.o memcached-bipbuffer.o memcached-logger.o memcached-crawler.o memcached-itoa_ljust.o memcached-slab_automove.o memcached-authfile.o memcached-restart.o memcached-proto_text.o memcached-proto_bin.o      memcached-extstore.o memcached-crc32c.o memcached-storage.o memcached-slab_automove_extstore.o 
Usage: dtrace [-aACeFHlqSvVwZ] [-arch i386|x86_64] [-b bufsz] [-c cmd] [-D name[=def]]
[snip]
```

(rebased on top of next and tried it, mac should be latest OS)

**Comment by devnexen** [comment]
It might need System Integrity disabled (at reboot time I think).

**Comment by dormando** [comment]
I've no idea what that is :( I was testing with a VM.

**Comment by dormando** [comment]
not sure why a system integrity thing would give a dtrace program syntax error though?

**Comment by devnexen** [comment]
The system integrity disallows dtrace to create object files. I found out however in the meantime you can "fine grain" the setting still enabling system integrity but fully enable dtrace capabilities.

**Comment by dormando** [comment]
@devnexen mind rebasing this too? I'd like to either merge or close it today. it's still necessary, yes?

**Comment by devnexen** [comment]
seems to be still needed.

**Comment by dormando** [comment]
@devnexen point it at `next` ? sorry!

**Comment by dormando** [comment]
nevermind I could do it, heh

**Comment by dormando** [comment]
@devnexen failing the whitespace test.

**Comment by dormando** [comment]
released in 1.6.17

**Pull request closed without merging**